### ISSUE TITLE
Media and other Sources

### ISSUE THREAD

**Issue by Hamath-Eltrael** [abdullahkhursheed22/factsilove#4]
Making a new issue for the Sources problem, to keep it organized. This started on #3 .

- [ ] Find Media Sources and credit(or replace) according to their license
- [ ] Find out if and what Texts are not original
- [ ] If non original texts exist, find sources and credit(or replace) according to their license

**Comment by Hamath-Eltrael** [comment]
If possible assign me to this and the License issue, so I don't overlook it. Thanks!

**Comment by abdullahkhursheed22** [comment]
Hey! So the texts are all original. I learned most of the information from video sources anyway, so I summarized and reworded it myself.

For the images used, it's either screenshots from videos or pictures from Google. I created the GIFs and one video clip by cropping YouTube videos. If it's not that big of an issue, we can skip this task since it will be a hassle for you.

**Comment by Hamath-Eltrael** [comment]
That's perfect for the text! 

And for all the Media, it is your decision. Let me tell you something though: depending on the licenses that the media are put under by their Creator it is _possible_ that they might send you a takedown notice, or worst case sue you and every who has a public fork. 

Or nothing might happen. That's why I offered myself to look into them. But I can also just not do that and we'll see what happens, if something ever happens.

**Comment by Hamath-Eltrael** [comment]
Then I'd have some time to improve and expand the content!

**Comment by abdullahkhursheed22** [comment]
> depending on the licenses that the media are put under by their Creator it is _possible_ that they might send you a takedown notice, or worst case sue you and every who has a public fork.

I didn't know that! In that case, it'd be better to be safe than sorry. If you have experience with this, then I would like to do everything to help you with that. Otherwise, should I also look into how to properly find sources and credit media as well?

**Comment by abdullahkhursheed22** [comment]
By the way, I am really impressed by the enthusiasm you have for the project and I couldn't be more thankful. Would you mind sharing what enabled such a mindset? I think I can learn from it and apply it in my own life.

**Comment by Hamath-Eltrael** [comment]
> By the way, I am really impressed by the enthusiasm you have for the project and I couldn't be more thankful. Would you mind sharing what enabled such a mindset? I think I can learn from it and apply it in my own life.

Well, I'd say it's the fact that there are just so many cool and half-random facts here! I love this and I also love learning new interesting stuff. I'd also love to contribute my knowledge, but I feel that this is the best I can do at the moment. But I'm hoping that we can make this to a kind of Wikipedia but for only the interesting stuff, easy to understand and connected visually. No overly technical language and partially like a kind of written VSauce I guess? I might just be my love for facts, that are easy to remember, understandable and cool.

**Comment by Hamath-Eltrael** [comment]
> > depending on the licenses that the media are put under by their Creator it is _possible_ that they might send you a takedown notice, or worst case sue you and every who has a public fork.
> 
> I didn't know that! In that case, it'd be better to be safe than sorry. If you have experience with this, then I would like to do everything to help you with that. Otherwise, should I also look into how to properly find sources and credit media as well?

`Warning: While we do this, we shouldn't add any new media to the project!`

I can do this. No problem. It's mostly a matter of plugging stuff into google Pictures and see what license the website uses. 

If you want to do stuff I could either:
-  Make new Issues for each media that's need replacing because of the license of them, and you could find new Media
- Alternatively do the same but as comments in this issue
- You could do the Plugging into Google(Pictures) stuff and make Issue for each here with checkboxes for Sites you got media from(This would be the greatest help, as I then just have to find the license and don't have to wait for my slow internet)
- Alternatively do the same but as comments in this issue

- The two option above together would make me a mere license finder, which is also good, if you have more free time than me, that you can invest in this.

It sadly does not really make sense to split one of those above.
If I could freely choose, this would be the way I'd like it to do:

1. You Plug the Pictures (images folder) into Google.
2. Make a new Issue for each new Site you took an image from, add a basic link to any page (preferably the homepage) to the first comment. Note the Name of each Picture in the subsequent comments(in case we need to replace them)
3. If there is already an Issue for the Site the image was taken from, don't create a new one.
4. I'll look at their License and make pull-request changes accordingly, I'll try to reference the Issue for these, so it's linked
5. I can do this while you are still adding more pictures to that Sites issue, for all that don't have to be taken down or need a special something for each single picture. Usually a: "We took pictures from ____ site, thank you." Should be enough. 
6. If the Picture(s) form that Site have to be taken down, I'll mark that in the Issue for that Site.
7. After I mark that, you _can_ (once create) and assign a label for that to that Sites Issues. 
8. For those that have to be changed, we'll do them all at once for each Site at the end, when all pictures have been googled. 
9. for that, we should try to do all from on site in one pull-request. Just to keep it organised
10. We can split these by Site if you assign the according Issues. 
11. DONE(I think)

**Comment by Hamath-Eltrael** [comment]
After we do this License Cleanup, we should make sure we don't have to do it again. For that we need a system to add Sources when adding Media.

There are two ways I'm aware of, at the moment:

- when adding Media add sources in the same pull-request/commits, if not already present for that site. 
- Create and Track it in the Issues(via Label:sources or something). This would probably be easier, as it is searchable. For this we can use the Issues we created for the cleanup. 1. You look if the source is already in an Issue, if not create on, if yes add filename to comments of that. 2. When not done for that issue, assign somebody for license creation. 3. Pull-request the issue with the license, or replace. 4. Close issue to mark that the License stuff has been done.

**Comment by Hamath-Eltrael** [comment]
We should try to keep the Site issues clean of all but the filenames in the comments, you can just reference them in a new/other issue. 

Also, if you need me to elaborate, do ask! I don't know how well I got my point across. I can also create examples, if you want.

**Comment by Hamath-Eltrael** [comment]
Oh and do let me know how we'll continue and if you have any other ideas or suggestions just say it!

**Comment by abdullahkhursheed22** [comment]
> but for only the interesting stuff, easy to understand and connected visually

I'm glad you like the facts! This is exactly why I wanted to create one in the first place. I'd learn about interesting facts every 20 pages of a book or every 30 minutes of a podcast. They were all over the place and there were no images to go along with them. So, I wanted to simplify the language and add pictures to make them appealing and easy to learn.

> 1. You Plug the Pictures (images folder) into Google.

I can try starting with this. I'm new to GitHub too but I'll watch a quick tutorial on creating and managing issues. For now, I was thinking of creating a list of media and its sources in a Google Sheet.

**Comment by abdullahkhursheed22** [comment]
> Shall I do an example Issue for one of the images?

Sure! That'd be very helpful.

**Comment by Hamath-Eltrael** [comment]
See #8 for example. Although it is a bad one, since it's not originally from that site. 
If you could [add a "Sources" Label](https://docs.github.com/de/issues/using-labels-and-milestones-to-track-work/managing-labels#creating-a-label) to that issue the example would be complete.

**Comment by Hamath-Eltrael** [comment]
And as you hopefully can see, we would create an Issue for each site that we took an issue from. We'd simply add another comment with witch file also is from that site.

**Comment by abdullahkhursheed22** [comment]
Hey! Thank you for explaining the process.

1. I created the "sources" label for issues on media sources.

2. I created two new issues as examples and would like your feedback. 

I searched for sources by uploading the files on Google Images. There were many different search results for each and I couldn't tell which ones were the original; so I selected the very first ones. I have attached an example of the search results I got below:

![image](https://github.com/abdullahkhursheed22/factsilove/assets/134539459/7e2c6cb7-17ee-4a20-9c2c-80e2c17a883f)

If I did everything okay, then I'll start creating issues on each media source for all the other files.

**Comment by abdullahkhursheed22** [comment]
Also, I don't know much about the next steps once all sources have been identified. So I'd also highly appreciate it if you could let me know how this information will be added to Obsidian and how it will help us integrate licenses. Thanks!

**Comment by Hamath-Eltrael** [comment]
#10 <--- is the wrong way round
#11 <--- this is looking good! Now if you'd have a second image from Shutterstock, you could just add them as another comment on #11

**Comment by abdullahkhursheed22** [comment]
Makes sense! I'll start doing this.

**Comment by Hamath-Eltrael** [comment]
Thank you very much. I'll get started with them ASAP. Hopefully tomorrow. 

Also, as I noticed that you deleted an issue, it's considered good _etiquette_ to not delete issues. I get why you might do it, but I'd recommend not to in the Future. Of course you can delete issues and comments that are downright offensive, but other than that, there probably isn't a reason. You can just close the issue with a comment staying the error. That leaves a better trace for future reference. there's other stuff you could do, but that's just overkill for us at the moment.

**Comment by abdullahkhursheed22** [comment]
Okay, that makes sense. I understand. Thank you for letting me know.

I did my bachelors in computer engineering but never got the chance to use GitHub for some reason. And I completely shifted my career direction after graduation. So I'm not familiar with best practices.

**Comment by Hamath-Eltrael** [comment]
Absolutely no problem! And just wanted to let you know.

**Comment by Hamath-Eltrael** [comment]
- [ ] You Plug the Pictures (images folder) into Google.
- [ ] Make a new Issue for each new Site you took an image from, add a basic link to any page (preferably the homepage) to the first comment. Note the Name of each Picture in the subsequent comments(in case we need to replace them)
- [ ] If there is already an Issue for the Site the image was taken from, don't create a new one.
- [ ] I'll look at their License and make pull-request changes accordingly, I'll try to reference the Issue for these, so it's linked
- [ ] I can do this while you are still adding more pictures to that Sites issue, for all that don't have to be taken down or need a special something for each single picture. Usually a: "We took pictures from ____ site, thank you." Should be enough.
- [ ] If the Picture(s) form that Site have to be taken down, I'll mark that in the Issue for that Site.
- [ ] After I mark that, you can (once create) and assign a label for that to that Sites Issues.
- [ ] For those that have to be changed, we'll do them all at once for each Site at the end, when all pictures have been googled. For that, we should try to do all from on site in one pull-request. Just to keep it organised
- [ ] We can split these by Site if you assign the according Issues.

**Comment by abdullahkhursheed22** [comment]
Understood! As you know, I'm working on the first three items right now.

**Comment by Hamath-Eltrael** [comment]
Yes, that's great! I'll try to do 4. as much as possible. But I just got a new job that has priority.

One more thin thought: please assign me to those issues, it should keep them on my radar and give me the ability to close them when they're done. See down the page for how to do multiples at once: https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users

**Comment by abdullahkhursheed22** [comment]
Congrats and best of luck on your new job!

It seems that I can assign issues to you that you have created, but not the ones I have created. It does not suggest your name for the issues I created:

![image](https://github.com/abdullahkhursheed22/factsilove/assets/134539459/c89a4407-c02a-49cc-92af-5632eb9b6c43)

**Comment by abdullahkhursheed22** [comment]
By the way, I'm pretty sure that the search results on Google Photos have gotten worse for some reason.

It keeps suggesting social media accounts and foreign sites instead of the actual sources. For example, I got some images from Wikipedia but it suggested someone's Twitter or Reddit account who reposted it.

I'm not sure what the issue is here.

**Comment by Hamath-Eltrael** [comment]
Hmmm, that'll make it more interesting... I just hope most are good. Those will be a bit to a lot more work. But we'll get to them then.

Sadly there's not much one can do to get the sources easier.

**Comment by Hamath-Eltrael** [comment]
But the one nice thing of license is you just have to get one from someone, you can assume that they are allowed to allow you usage or will otherwise tell you who to ask. (As far as I know a believe)

**Comment by abdullahkhursheed22** [comment]
Okay, makes sense!

And how do I resolve this problem of assigning issues to you:

> It seems that I can assign issues to you that you have created, but not the ones I have created. It does not suggest your name for the issues I created.

**Comment by Hamath-Eltrael** [comment]
Ok, so I'll put a comment on the ones I started doing something and then you can assign me.

**Comment by abdullahkhursheed22** [comment]
Okay! I have created issues by adding the sources for all media. I look forward to hearing from you on how I can help with the next steps. @Hamath-Eltrael

**Comment by abdullahkhursheed22** [comment]
Hey! Since it's been a while, I was thinking I should keep adding more content to Obsidian and I'll create issues for whatever media I add along the way.

**Comment by Hamath-Eltrael** [comment]
Yes, I’m Very Sorry. The Life is getting in the way. But yes. That’ll work.

**Comment by abdullahkhursheed22** [comment]
No worries at all! I completely understand and you're doing a huge favor. Please take your time.
